jupyterlab/commands.py

```diff
@@ -108,18 +108,34 @@ def dedupe_yarn(path, logger=None):
         yarn_proc.wait()
 
 
-def ensure_dev(logger=None):
-    """Ensure that the dev assets are available.
+def ensure_node_modules(cwd, logger=None):
+    """Ensure that node_modules is up to date.
+
+    Returns true if the node_modules was updated.
     """
-    parent = pjoin(HERE, '..')
+    logger = _ensure_logger(logger)
+    yarn_proc = Process(['node', YARN_PATH, 'check', '--verify-tree'], cwd=cwd, logger=logger)
```

Comment:
Neat! For future ref: This command completes in a little less than 5 seconds on my system, which should be OK for the code paths this gets invoked from (app start for core/dev/watch mode only).